examples/api/lib/widgets/editable_text/editable_text.on_content_committed.0.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for EditableText.onContentCommitted
+
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: MyStatefulWidget(),
+    );
+  }
+}
+
+class MyStatefulWidget extends StatefulWidget {
+  const MyStatefulWidget({super.key});
+
+  @override
+  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
+}
+
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  final TextEditingController _controller = TextEditingController();
+  Uint8List? bytes;
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          const Text("Here's a text field that supports inserting only png or gif content:"),
+          TextField(
+            controller: _controller,
+            contentCommitMimeTypes: const <String>['image/png', 'image/gif'],
+            onContentCommitted: (CommittedContent data) async {
+              if (data.data != null) {
+                setState(() {
+                  bytes = data.data;
+                });
+              }
+            },
+          ),
+          if (bytes != null)
+            const Text("Here's the most recently inserted content:"),
+          if (bytes != null)
+            Image.memory(bytes!),
+        ],
+      ),
+    );
+  }
+}
```

Comment:
One more thing that we should probably do actually: Can you write a test for this example?  Maybe pump this app, send the commitContent message similar to how you do that in your editable_text_test.dart test, and then expect that the Image widget appears?

Here's an example of testing something in the examples directory for reference:
https://github.com/flutter/flutter/blob/master/examples/api/lib/widgets/heroes/hero.0.dart
https://github.com/flutter/flutter/blob/master/examples/api/test/widgets/heroes/hero.0_test.dart

Reply:
I'm not able to get this going... it seems that the input client is either ignoring or not getting notified about the mock commit content invocation?

This is the code:

```dart
testWidgets('Image.memory displays committed content', (WidgetTester tester) async {
    await tester.pumpWidget(
      const example.ContentCommittedApp(),
    );

    expect(find.text('Content Committed Sample'), findsOneWidget);

    const String uri = 'content://com.google.android.inputmethod.latin.fileprovider/test.png';
    const List<int> kBlueSquarePng = <int>[
      0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a, 0x00, 0x00, 0x00, 0x0d, 0x49,
      0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x32, 0x00, 0x00, 0x00, 0x32, 0x08, 0x06,
      0x00, 0x00, 0x00, 0x1e, 0x3f, 0x88, 0xb1, 0x00, 0x00, 0x00, 0x48, 0x49, 0x44,
      0x41, 0x54, 0x78, 0xda, 0xed, 0xcf, 0x31, 0x0d, 0x00, 0x30, 0x08, 0x00, 0xb0,
      0x61, 0x63, 0x2f, 0xfe, 0x2d, 0x61, 0x05, 0x34, 0xf0, 0x92, 0xd6, 0x41, 0x23,
      0x7f, 0xf5, 0x3b, 0x20, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x36, 0x06, 0x03, 0x6e, 0x69, 0x47, 0x12, 0x8e, 0xea, 0xaa,
      0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82,
    ];
    final ByteData? messageBytes = const JSONMessageCodec().encodeMessage(<String, dynamic>{
      'args': <dynamic>[
        -1,
        'TextInputAction.commitContent',
        jsonDecode('{"mimeType": "image/png", "data": $kBlueSquarePng, "uri": "$uri"}'),
      ],
      'method': 'TextInputClient.performAction',
    });

    try {
      await ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
        'flutter/textinput',
        messageBytes,
            (ByteData? _) {},
      );
    } catch (_) {}

    await tester.pump();
    final Size imgSize = tester.getSize(find.byType(Image));
    expect(imgSize.width.roundToDouble(), 50.0);
    expect(imgSize.height.roundToDouble(), 50.0);
  });
```

The `onContentCommitted` callback of the text field is never executed.